src/cclib/parser/orcaparser.py

```diff
@@ -934,6 +935,229 @@ def energy_intensity(line):
                 polarizability.append(line.split())
             self.polarizabilities.append(numpy.array(polarizability))
 
+        if line.strip() == 'ORCA-CASSCF':
+            self.skip_lines(inputfile, ['d', 'b'])
+            # -------------------------------------------------------------------------------
+            #                               ORCA-CASSCF
+            # -------------------------------------------------------------------------------
+            #
+            # Symmetry handling      UseSym     ... ON
+            # Point group                       ... C2
+            # Used point group                  ... C2
+            # Number of irreps                  ... 2
+            #    Irrep    A has   10 SALCs (ofs=   0) #(closed)=   0 #(active)=   2
+            #    Irrep    B has   10 SALCs (ofs=  10) #(closed)=   0 #(active)=   2
+            #  Symmetries of active orbitals:
+            #    MO =    0  IRREP= 0 (A)
+            #    MO =    1  IRREP= 1 (B)
+            vals = next(inputfile).split()
+            if vals[0] == 'Symmetry':
+                # Symmetry section is only printed if symmetry is used
+                assert vals[-1] == 'ON'
+                point_group = next(inputfile).split()[-1]
+                used_point_group = next(inputfile).split()[-1]
+                num_irreps = int(next(inputfile).split()[-1])
+                num_active = 0
+                for i, line in zip(range(num_irreps), inputfile):
+                    reg = r'Irrep\s+(\w+) has\s+(\d+) SALCs \(ofs=\s*(\d+)\) #\(closed\)=\s*(\d+) #\(active\)=\s*(\d+)'
```

Comment:
How about using `re.compile` outside of the for-loop so it only has to be created once?

Reply:
Compiling doesn't always result in a significant speedup due to caching done by the interpreter ([compiling python regex](https://stackoverflow.com/questions/452104/is-it-worth-using-pythons-re-compile)). I imagine the overhead of keeping track of the regexes may make this code more difficult to maintain. Furthermore, this parses 10 lines at most, so looking for a potential speedup in this section of the code is like lipstick on a pig.
Of course, hard results are much more useful. Thus I tested what I expected to be the section of the code that can most benefit from compiled regexes. On 50 roots of chemists water, the [non-compiled ran faster](https://pastebin.com/h0xXzzVw)!? Python does a lot of magic behind the scenes with caching. I tried with a variety of different input, and the compiled version was consistently slower (though only marginally).
Barring some error in my testing technique (very possible), I think that compiling regular expressions only makes the coding more difficult. Any speedup that could be had is probably inconsequential compared to the rest of the code.

Reply:
Also, from the official python documentation on [re.compile](https://docs.python.org/3.5/library/re.html#re.compile).
"Note: The compiled versions of the most recent patterns passed to re.compile() and the module-level matching functions are cached, so programs that use only a few regular expressions at a time needn’t worry about compiling regular expressions."

Reply:
That's really good to know. I didn't realize that patterns were cached by module-level functions.

For your test, perhaps the reason the non-compiled versions ran faster was because you used `re.search` for both string regex and compiled regex instead of using the `search()` function of the compiled regex. For example:

```
prog = re.compile(pattern)
result = prog.search(string)
```
 instead of 

```
prog = re.compile(pattern)
result = re.search(prog, string)
```

Of course, those may be equivalent behind-the-scenes.

Regardless, my suggestion for compiling the regex and place outside of the loop was also about code style (i.e only variable objects in the for-loop), but I don't feel strongly about that since in this case, putting the regex string in the loop is to make the code easier to read.